deepspeed/__init__.py

```diff
@@ -58,11 +58,12 @@ def initialize(args=None,
                mpu=None,
                dist_init_required=None,
                collate_fn=None,
-               config_params=None):
+               deepspeed_config=None):
```

Comment:
Sorry, I think we have had a bit of a miscommunication in https://github.com/microsoft/DeepSpeed/pull/983

This would be a breaking change.

I think it'd be OK for `transformers` since we constantly raise the minimal requirement for deepspeed, but is this the case for other DeepSpeed users?

Reply:
Had a feeling, lots of concepts were discussed :)

If we like the change but mind BW compat, I can re-add the `config_params` parameter, setting it to `deepspeed_config` (optionally I'll print a warning message, if we like).

Reply:
My feeling is that they haven't used `deepspeed_` prefix in the first place, because it's all deepspeed and thus doesn't require self-reference ;)

Reply:
haha that's a great point! maybe `config_params` should be kept as is then? Just internally we've renamed?

I'd prefer just `config` if I had to choose though

Reply:
Yes, config params repeats itself twice.

I agree `config` would be the cleanest.

Well, since the deepspeed trusts us to mess with their API, I hope it won't be overstepping if you made it `config` and left `config_params` as a back-compat deprecated arg name. 

But personally I'm totally fine with `config_params` since it's only one place where it's used.

Internally there is no back-compat issue I guess, so if it makes the code more intuitive then it's probably a good choice.

Reply:
so with that said, `config` will be the new name, and `config_params` will remain as BW compat, let's see if @tjruwase is cool with it!

Reply:
Thanks guys for driving this conversation. Yes, we have avoided using `deepspeed_` because of the self-reference. I like the proposal of `config` for new name and `config_params` for BW compat. I will also ping the rest of the folks here for awareness of the changes.

Reply:
I am also a fan of `config` for the new name!

Reply:
Sounds good to me as well :)